x/ibc/04-channel/keeper/keeper.go

```diff
@@ -138,18 +156,10 @@ func (k Keeper) GetPacketAcknowledgement(ctx sdk.Context, portID, channelID stri
 	return bz, true
 }
 
-// IteratePacketSequence provides an iterator over all send and receive sequences. For each
-// sequence, cb will be called. If the cb returns true, the iterator will close
-// and stop.
-func (k Keeper) IteratePacketSequence(ctx sdk.Context, send bool, cb func(portID, channelID string, sequence uint64) bool) {
-	store := ctx.KVStore(k.storeKey)
-	var iterator db.Iterator
-	if send {
-		iterator = sdk.KVStorePrefixIterator(store, []byte(host.KeyNextSeqSendPrefix))
-	} else {
-		iterator = sdk.KVStorePrefixIterator(store, []byte(host.KeyNextSeqRecvPrefix))
-	}
-
+// IteratePacketSequence provides an iterator over all send, receive or ack sequences.
+// For each sequence, cb will be called. If the cb returns true, the iterator
+// will close and stop.
+func (k Keeper) IteratePacketSequence(ctx sdk.Context, iterator db.Iterator, cb func(portID, channelID string, sequence uint64) bool) {
```

Comment:
Since you are passing in an iterator now, it may be an iterator over anything in state. Thus, we should be checking that the key and value has the format we expect. Namely that keySplit has atleast 5 elements. Otherwise this function can panic and crash the chain if an unexpected iterator gets passed in